staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -173,8 +174,12 @@ func fromUnstructured(sv, dv reflect.Value) error {
 					dv.Set(sv.Convert(dt))
 					return nil
 				}
+				if sv.Float() == math.Trunc(sv.Float()) {
```

Comment:
Done (unfortunately I should have done it explicitly, to force the map[string]interface{] to really contain float).